forte/processors/allennlp_processors.py

```diff
@@ -0,0 +1,145 @@
+                               '`allow_parallel_entries` are True, all '
+                               'existing conflicting entries will be deleted.')
+        else:
+            if not self.allow_parallel_entries:
+                logger.warning('Both `overwrite_entries` and '
+                               '`allow_parallel_entries` are False, '
+                               'processor will only run if there are no '
+                               'existing conflicting entries.')
+
+    @staticmethod
+    def default_configs():
+        """
+        This defines a basic Hparams structure for AllenNLP.
+        :return:
+        """
+        return {
+            'processors': "tokenize,pos,depparse",
+            # format of the POS tags and dependency parsing
+            'output_format': "universal_dependencies",
+            # whether to overwrite the entries of the same Type as produced by
+            # this processor
+            'overwrite_entries': True,
+            # whether to allow similar new entries when they already exist, e.g.
+            # allowing new tokens with same spans, used only when
+            # `overwrite_entries` is False
+            'allow_parallel_entries': False
+        }
+
+    def _process(self, input_pack: DataPack):
+        # handle existing entries
+        self._process_existing_entries(input_pack)
+
+        for sentence in input_pack.get(Sentence):
+            result = self.predictor.predict(sentence=sentence.text)
+
+            if "tokenize" in self.processors:
+                # creating new tokens and dependencies
+                tokens = self._create_tokens(input_pack, sentence, result)
+                if "depparse" in self.processors:
+                    self._create_dependencies(input_pack, tokens, result)
+
+    def _process_existing_entries(self, input_pack):
+        tokens_exist = any(True for _ in input_pack.get(Token))
+        dependencies_exist = any(True for _ in input_pack.get(Dependency))
+
+        if tokens_exist or dependencies_exist:
+            if not self.overwrite_entries:
+                if not self.allow_parallel_entries:
+                    raise ProcessorConfigError("Found existing entries, either "
+                                               "`overwrite_entries` or "
+                                               "`allow_parallel_entries` "
+                                               "should be True")
+            else:
+                # delete existing tokens and dependencies
+                for entry_type in (Token, Dependency):
+                    for entry in input_pack.get(entry_type):
+                        input_pack.delete_entry(entry)
+
+    def _create_tokens(self, input_pack, sentence, result):
+        words, pos = result['words'], result['pos']
+        tokens = []
+        offset = sentence.span.begin
+        word_end = 0
+        for i, word in enumerate(words):
+            word_begin = sentence.text.find(word, word_end)
```

Comment:
No this won't work. This will give one the first occurrence. Once there are two same tokens, this will fail.

Reply:
I saw this technique being used in all our existing tokenizers (spacy, nltk, stanfordnlp).  
It's checking the first occurrence after the ending of previous token. Can you give an example where it won't work?

Reply:
I see, didn't notice that. I am always trying to use the offset from the original method and ignores this way.  Now I find probably the main drawback is the O(n^2) complexity, where I think an O(n) solution probably exists. We can make a utility function that does O(n) for similar situations and use them later.

Reply:
I agree, had discussed this too with @AvinashBukkittu. The problem was that the offset method used in out readers assumes that the delimiter is a space. Here this might not be the case.
I agree an O(n) solution probably exists though which can take care of any delimiter. 

Here though we are not searching the whole sentence anytime, we start with the ending of previous token and the word will be found just after the delimiter so in effect the complexity should still be O(n) I think.

Reply:
I don't know whether changing the delimiter would falsify this algorithm. But yes, I think there is a more universal solution to handle this.